README.md

```diff
@@ -2,63 +2,101 @@
-- In order to use `GigaSOM`, you need to install Julia 1.0 or higher. You can find the download and installation instructions for Julia [here](https://julialang.org/downloads/).
+- **Operating system**: Use Linux (Debian, Ubuntu or centOS), MacOS, or Windows 10 as your operating system. GigaSOM has been tested on these systems.
+- **Julia language**: In order to use GigaSOM, you need to install Julia 1.0 or higher. You can find the download and installation instructions for Julia [here](https://julialang.org/downloads/).
+- **Hardware requirements**: GigaSOM runs on any hardware that can run Julia, and can easily use resources from multiple computers interconnected by network. For processing large datasets, you require to ensure that the total amount of available RAM on all involved computers is larger than the data size.
 
-Once `julia` has been installed, follow [these
-instructions](https://docs.julialang.org/en/v1/manual/getting-started/) to get
-started.
-
-:bulb: If you are a complete beginner with Julia, it is advised that you familiarize youself
-with the [full documentation](https://docs.julialang.org). You should also have a
-closer look at the [Julia package manager](https://julialang.github.io/Pkg.jl/v1/getting-started/).
+:bulb: If you are new to Julia, it is adviseable to [familiarize youself with
+the environment
+first](https://docs.julialang.org/en/v1/manual/getting-started/).  Use the full
+Julia [documentation](https://docs.julialang.org) to solve various possible
+language-related problems, and the [Julia package manager
+docs](https://julialang.github.io/Pkg.jl/v1/getting-started/) to solve
+installation-related difficulties.
 
 ## Installation
 
-At the Julia prompt, add the `GigaSOM` package:
+Using the Julia package manager to install GigaSOM is easy -- after starting Julia, type:
 
 ```julia
-julia> import Pkg; Pkg.add("GigaSOM");
+import Pkg; Pkg.add("GigaSOM");
 ```
 
-Use the `GigaSOM` module by running:
+Then you can import the GigaSOM package and start using it:
 
 ```julia
-julia> using GigaSOM
+using GigaSOM
 ```
 
-## Test the installation
+The first "import" of the GigaSOM package may take several minutes to complete due to precompilation of the sources, especially on a fresh Julia install.
+
+### Testing the installation
 
-`GigaSOM` has been tested on Linux (Ubuntu and centOS), macOS, and Windows.
+If you run a non-standard platform (e.g. a customized operating systems), or if you added any modifications to GigaSOM source code, you may want to run the test suite to ensure that everything works as expected:
 
-You may test the package as follows:
+```julia
+import Pkg; Pkg.test("GigaSOM");
+```
 
+For debugging, it is sometimes very useful to enable the `@debug` messages from the source, as such:
 ```julia
-julia> import Pkg; Pkg.test("GigaSOM");
+using Logging
+global_logger(ConsoleLogger(stderr, Logging.Debug))
 ```
 
-Alternatively, you can use the package manager by hitting `]`:
+## How to use GigaSOM
```

Comment:
I'd suggest to put this into a tutorial, and then refer to that tutorial. The README should be concise IMO, and right now, it is too long.

Reply:
I wanted to have this really visible, pictures give people some sense of that trying the package is worth it.

(it really helps if you just scroll down and see the code + resulting pictures, ideally without leaving the page, or at least without leaving gthub. In ShinySOM I have a nice TUTORIAL.md with pictures, but 90% people don't even look at that.)